spacy/cli/_util.py

```diff
@@ -221,24 +221,45 @@ def validate_project_commands(config: Dict[str, Any]) -> None:
 
     config (Dict[str, Any]): The loaded config.
     """
+
+    def verify_workflow_step(workflow_name: str, step: str) -> None:
```

Comment:
We shouldn't let this depend on the definition of `command_names` that is below it - it makes the code less readible and more prone to errors. 

Instead, I propose to make this a private (non-nested) function that takes in `config` and `step`, or alternatively `workflow`, `commands`, and `step` (shorter variable names help with readability, too)